src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java

```diff
@@ -0,0 +1,120 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import edu.ucsb.cs156.gauchoride.entities.Ride;
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+import edu.ucsb.cs156.gauchoride.repositories.RideRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+
+@Api(description = "Ride Request")
+@RequestMapping("/api/ride_request")
+@RestController
+
+public class RideController extends ApiController {
+
+    @Autowired
+    RideRepository rideRepository;
+
+    @ApiOperation(value = "List all rides")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Ride> allRides() {
+        Iterable<Ride> rides = rideRepository.findAll();
+        return rides;
+    }
+
+    @ApiOperation(value = "Get a single ride")
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
See the code here for an example of how you can prevent a user from accessing the ride of another user.

https://github.com/ucsb-cs156/proj-courses/blob/644c5c7800fda04a14e053bd052df3cd7b60f8d7/src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java#LL58C1-L58C1

Another way to do it is get the ride, and then if the userid of the current user doesn't match the riderId for the ride, you throw a non-found exception.   Note that you could also throw a 403 (not authorized) but that leaks some information about whether the ride exists or not.

I'm not suggesting that the existence (or not) of a ride is sensitive information, but security best practice is to try to not leak anything that could help a hacker (and even though in this app, that isn't critical, it's good practice to try to implement the most secure solution.)